pkg/network/ifaceutil/interface.go

```diff
@@ -161,24 +197,67 @@ func MakeVethPair(name, master string, mtu int) (netlink.Link, error) {
 	}()
 
 	peerLink, err := netlink.LinkByName(peer)
-	if err = netlink.LinkSetMaster(peerLink, masterLink); err != nil {
-		return nil, err
+	if err != nil {
+		return nil, fmt.Errorf("failed to get peer link %q: %v", peer, err)
 	}
 
-	// make sure the lowerhalf is up, this automatically sets the upperhalf UP
-	if err = netlink.LinkSetUp(peerLink); err != nil {
-		return nil, errors.Wrap(err, "could not set veth peer up")
+	if err := netlink.LinkSetMaster(peerLink, masterLink); err != nil {
+		return nil, fmt.Errorf("failed to set master for peer link %q: %w", peer, err)
 	}
 
 	// Re-fetch the link to get its creation-time parameters, e.g. index and mac
-	veth2, err := netlink.LinkByName(name)
+	// either from the host, or from the namespace
+	var veth2 netlink.Link
+
+	veth2, err = netlink.LinkByName(name)
 	if err != nil {
 		return nil, err
 	}
 
+	if netNs != nil {
+		veth2, err = moveLinkToNamespace(veth, tmpName, name, netNs, peerLink)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	// make sure the lowerhalf is up, this automatically sets the upperhalf UP
+	if err := netlink.LinkSetUp(peerLink); err != nil {
+		return nil, fmt.Errorf("failed to set veth peer up: %w", err)
+	}
+
```

Comment:
I don't believe this function should return anything because the `veth` can be inside a namespace. So returning a link that is inside a namespace does not make any sense. Normally to work with links that lives inside a namespace you have to do 
```go
ns.Run(func() {
  // load the nic here
  // work with the nic
}
```